Remove CMake variables that cannot be changed in CCMake

rianquinn opened the pull request:
This patch makes sure that the ccmake tool only shows variables that in
theory should be able to be changed prior to running make.

[ISSUE]: https://github.com/Bareflank/hypervisor/issues/587

Signed-off-by: “rianquinn” <“rianquinn@gmail.com”>

JaredWright commented:
Only a couple comments:

1) The commit message uses the word cleanup

2) Are you sure we want to mark the toolchain configurations as advanced? This might make cross-compiling for ARM tricky

3) ccmake does indeed appear to be working on my environment, and the CMakeCache is being updated correctly as well. Was the root cause from using the add_config macro to define CMAKE_* variables?

👍 for getting ccmake working again!

rianquinn commented:
@JaredWright 

1: Fixed
2: Advanced only effects ccmake. If your going to change a toolchain, it needs to be done on the command line. That is a cmake limitation
3. Awesome. That fix I added was only there to fix an issue with VERBOSE not having a default value. To be honest, I'm not sure what changed as it was working fine when I tried it (outside of the variables that you cannot change)

Your welcome 😄
